### PR TITLE
[7.x] Convert eloquent collection to eloquent query builder

### PR DESCRIPTION
**Pull request by ahmedsayedabdelsalam**
Some times we have a collection of models that we need to update all of them at once

I found the regular approach is to loop on every single model and make (N) times update query

I can build new query from scratch to update them all something like `Model::whereIn('id', $items->pluck('id'))->update([])`
 but I think writing it this way is ugly what if we can convert this collection of models to a query builder looking for those models under the hood.
so we can write it like that `$items->toQuery()->update([])`

the following image is from a project I am working on it. I am using it as a macro and I thought it could be useful if added to the framework.

Screen Shot 2020-06-27 at 12 14 32 AM

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
This won't work in general. If there are lots of items in the collection, you will bust the maximum query length limit of any database.

File: src/Illuminate/Database/Eloquent/Collection.php [hunk 1]
```diff
@@ -629,4 +629,22 @@ public function getQueueableConnection()
...
+     */
+    public function toQuery()
+    {
+        $model = $this->first();
+
+        if (! $model) {
+            throw new LogicException('can not get Eloquent QueryBuilder from an empty Collection.');
+        }
+
+        return $model->newModelQuery()->whereKey($this->modelKeys());
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Collection.php:648]: I assume you wanted "where in" and not "where". Have you tried this code for real?

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Database/Eloquent/Collection.php (hunk 1 above)
Comment (ahmedsayedabdelsalam) [reply src/Illuminate/Database/Eloquent/Collection.php:648]: Screen Shot 2020-06-27 at 1 04 03 AM

yes I tried it and **loadCount** function is implemented the same way

**Comment by ahmedsayedabdelsalam** [comment]
> This won't work in general. If there are lots of items in the collection, you will bust the maximum query length limit of any database.

is `Model::update([]);` will not work either ?

**Pull request merged**